services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/team/MembershipRequestManagerImpl.java

```diff
@@ -103,6 +104,10 @@ public MembershipRqstSubmission create(UserInfo userInfo,
 			if (hasUnmetAccessRequirements(userInfo, mrs.getTeamId()))
 				throw new UnauthorizedException("Requested member has unmet access requirements which must be met before asking to join the Team.");
 		}
+		Team team = teamDAO.get(mrs.getTeamId());
+		if (team.getCanPublicJoin() != null && team.getCanPublicJoin()) {
+			throw new IllegalArgumentException("Cannot invite to public team.");
```

Comment:
Same here, we should explain that the team is open for public to join. Therefore, user doesn't have to create a request.